### ISSUE TITLE
False positive undefined-variable

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#3434]
### Steps to reproduce
```python
import collections

l = ["a","b","c"]

class Foo(collections.namedtuple("Foo",[x+"_foo" for x in l])):
    pass
```

### Current behavior

> error (E0602, undefined-variable, Foo) Undefined variable 'x'

for the `class Foo` line

### Expected behavior

no diagnostics - the code works fine

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.7.6 (default, Dec 30 2019, 19:38:28) 
[Clang 11.0.0 (clang-1100.0.33.16)]
```

**Comment by Pierre-Sassoulas** [comment]
Fixed in #3713
